frontend/javascript/products.js

```diff
@@ -0,0 +1,82 @@
+import { fetchData } from './fetch.js';
+
+function handleProductError() {
+    const fourPlaceholderProducts = 4;
+    for (let i = 0; i < fourPlaceholderProducts; i++) {
+        if (doesBrowserSupportTemplete) {
+            const templateClone = createTemplate();
+            let prodImage = templateClone.querySelector('#trending-div-product-image');
+            prodImage.setAttribute('src', './assets/images/trending-product.webp');
+            prodImage.setAttribute('alt', 'Image of a trending product');
+            addToProductsList(templateClone);
+        }
+    }
+}
+
+const apiUrl = 'http://localhost:8080/products/';
+
+window.onload = (event) => {
+    fetchData(apiUrl, appendProducts).then(appendProducts).catch(handleProductError);
+};
+
+function createProductImage(template) {
+    let prodImage = template.querySelector('#trending-div-product-image');
+    // Need to test for null image once image is added to API
+    prodImage.setAttribute('src', './assets/images/trending-product.webp');
+    prodImage.setAttribute('alt', 'Image of a trending product');
+}
+
+function createProductLabel(template, product) {
+    let labelDiv = template.querySelector('#trending-div-product-label');
+    if (product.name == null) {
```

Comment:
`name` and `unitPrice` may be `null`, but if they don't exist as properties in the response at all they will be `undefined`. One alternative is to check `if (!product.name)` instead. 

An edge case though is when the name is an empty string, or the price is 0 - in that case the "falsy" condition will be fulfilled as well, so perhaps we we could do `if (typeof test === 'undefined' || test === null)`?